feature/dashboard/src/main/java/com/simprints/feature/dashboard/settings/syncinfo/usecase/ObserveSyncInfoUseCase.kt

```diff
@@ -227,13 +236,15 @@ internal class ObserveSyncInfoUseCase @Inject constructor(
             isCounterImagesToUploadVisible = isPreLogoutUpSync,
             counterImagesToUpload = imagesToUpload?.toString().orEmpty(),
             isInstructionDefaultVisible = !isModuleSelectionRequired &&
-                isConnected &&
+                isEventSyncConnectionSufficient &&
                 !eventSyncState.isSyncFailed() &&
                 !eventSyncState.isSyncInProgress() &&
                 !isPreLogoutUpSync,
-            isInstructionNoModulesVisible = isConnected && isModuleSelectionRequired && !isEventSyncInProgress,
-            isInstructionOfflineVisible = !isConnected,
-            isInstructionErrorVisible = isConnected && eventSyncState.isSyncFailed(),
+            isInstructionCommCarePermissionVisible = isCommCarePermissionMissing,
+            isInstructionNoModulesVisible = isEventSyncConnectionSufficient && !isCommCarePermissionMissing && isModuleSelectionRequired
```

Comment:
Why is `isCommCarePermissionMissing` part of the logic for `isInstructionNoModulesVisible`?